tools/targets/PSOC6.py

```diff
@@ -133,5 +151,345 @@ def find_cm0_image(toolchain, resources, elf, hexf, hex_filename):
+    # reserve name for separate NSPE image
+    out_cm4_hex = binf[:-4] + "_cm4.hex"
+
+    # preserve original hex file from mbed-os build
+    mbed_hex = binf[:-4] + "_unsigned.hex"
+    copy2(binf, mbed_hex)
+
+    # find target name and type before processing
+    for part in PurePath(binf).parts:
+        if "CY" in part:
+            target_sig_data = process_target(toolchain=toolchain, target=part)
+
+    if target_sig_data is None:
+        toolchain.notify.debug("[PSOC6.sign_image] ERROR: Target not found!")
+        raise AddSignatureError("PSOC6.sign_image finished execution with errors! Signature is not added.")
+
+    for slot in target_sig_data:
+        # first check if image for slot under processing should be encrypted
+        if slot["slot_data"].get("encrypt") is True:
+            # call encrypt_img to perform encryption
+            args = [sys.executable, str(slot["sdk_path"] / "encrypted_image_runner.py"),
+                                    "--sdk-path", str(slot["sdk_path"]), "--hex-file", os.getcwd() + '/' + mbed_hex,
+                                    "--key-priv", str(slot["sdk_path"] / slot["key_file"]),
+                                    "--key-pub", str(slot["sdk_path"] / slot["dev_pub_key"]),
+                                    "--key-aes", str(slot["sdk_path"] / slot["aes_key"]),
+                                    "--ver", str(slot["img_data"]["VERSION"]), "--img-id", str(slot["id"]),
+                                    "--rlb-count", str(slot["img_data"]["ROLLBACK_COUNTER"]),
+                                    "--slot-size", str(hex(slot["slot_data"]["size"])),
+                                    "--img-offset", str(slot["slot_data"]["address"])]
+            if slot["slot_data"]["type"] != "BOOT":
+                args.append("--pad")
+            process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+            # catch standard process pipes outputs
+            stderr = process.communicate()[1]
+            stdout = process.communicate()[0]
+            rc = process.wait()
+            toolchain.notify.info(stdout.decode("utf-8"))
+
+            if rc != 0:
+                toolchain.notify.debug("[PSOC6.sign_image] ERROR: Encryption script ended with error!")
+                toolchain.notify.debug("[PSOC6.sign_image] Message from encryption script: " + stderr.decode("utf-8"))
+                raise AddSignatureError("PSOC6.sign_image finished execution with errors! Signature is not added.")
+            else:
+                toolchain.notify.info("[PSOC6.sign_image] SUCCESS: Image for slot " +
+                                    slot["slot_data"]["type"] + " is signed and encrypted with no errors!")
+        # all non ecrypted images take this path
+        else:
+            if slot["slot_data"]["type"] == "UPGRADE":
+                out_hex_name = binf[:-4] + "_upgrade.hex"
+                out_bin_name = out_hex_name[:-4] + "_signed.bin"
```

Comment:
Thanks for the changes, just one left. This can be moved outside the if then else block.

Reply:
Thank you for comment. This fixed in latest commit.